src/lkl/setup.c

```diff
@@ -636,6 +649,18 @@ static void lkl_run_in_kernel_stack(void* (*start_routine)(void*), void* arg)
     }
 }
 
+static bool is_encrypted_cfg(disk_config_t* cfg)
+{
+    return cfg->key || cfg->key_id || cfg->fresh_key;
+}
+
+#ifdef DEBUG
+static bool is_encrypted(sgxlkl_enclave_mount_config_t* cfg)
+{
+    return cfg->key || cfg->key_id || cfg->fresh_key;
+}
+#endif
```

Comment:
Why do we need to have two types? This doesn't seem elegant.

Reply:
No, this is not elegant at all, this is purely legacy structure. `disk_config_t` is a local type and there are more than 1 code points where it wants to check whether the disk is encrypted. The second one is used exclusively for debug output, but it could move to sgxlkl_enclave_config.h/.c.